### PR TITLE
chore: Cleannup + Automate releases

### LINKED ISSUE #43: Update AppVersion when new ParadeDB version is published
**Is your feature request related to a problem? Please describe.**
Currently, the value is set to the latest version, but will go out of sync once a new version of ParadebDB is published.

**Describe the solution you'd like**
Update the Helm Chart `AppVersion` and `imageName` default value whenever a new version is published.

This will involve committing to this repo from the main repository when a new release is published.

### PR DESCRIPTION
**Pull request by philippemnoel**
**Ticket(s) Closed**

- Closes #43

**What**
This PR lifts up the repository to be up to the same standards as the monorepo, paradedb/paradedb. It also automates the process of generating new releases for the ParadeDB helm chart, to make our lives easier.

**Why**

**How**

**Tests**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,36 +67,39 @@ for more information.
...
 
-    helm delete my-db
+```bash
+helm delete <my-db>
+```
 
 ## Configuration
 
-The ParadeDB Helm chart can be configured using the values.yaml file or by
+The ParadeDB Helm chart can be configured using the `values.yaml`` file or by
```
Comment (MauAraujo) [inline README.md:93]: There is an extra quote in `values.yaml`

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          - "eslint-config-prettier@^9.0.0"
+          - "eslint-plugin-prettier@^5.0.0"
+          - "eslint-plugin-unused-imports@^3.0.0"
+          - "prettier@^3.0.3"
+
+  - repo: https://github.com/doublify/pre-commit-rust
+    rev: v1.0
+    hooks:
+      - id: fmt
+      - id: clippy
```
Comment (MauAraujo) [inline .pre-commit-config.yaml:76]: do we need the rust linters too?

File: .github/workflows/publish-helm-chart.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+# the creation of a GitHub Release and only happens once we trigger a workflow_dispatch
+# event, either manually or via creating a release in the `paradedb/paradedb` repository.
+
+name: Publish Helm Chart
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
```
Comment (MauAraujo) [inline .github/workflows/publish-helm-chart.yml:14]: Question, if I understand correctly, when a new ParadeDB version is released, it triggers this workflow and both `appVersion` and `chartVersion` are incremented. But if this workflow is triggered by a push to main, only the `chartVersion` is incremented, and `appVersion` stays the same?

Comment (philippemnoel) [reply .github/workflows/publish-helm-chart.yml:14]: Correct. What would we increment appVersion to if it hasn't changed?

File: .pre-commit-config.yaml (hunk 2 above)
Comment (philippemnoel) [reply .pre-commit-config.yaml:76]: They don't run, but you're right I'll just remove them

File: README.md (hunk 1 above)
Comment (philippemnoel) [reply README.md:93]: Nice catch

File: .github/workflows/publish-helm-chart.yml (hunk 3 above)
Comment (MauAraujo) [reply .github/workflows/publish-helm-chart.yml:14]: No we shouldn't increment if it hasn't changed, it is tied to the ParadeDB versions. Thanks for explaining

Comment (philippemnoel) [reply .github/workflows/publish-helm-chart.yml:14]: Sweet :)

File: .pre-commit-config.yaml (hunk 2 above)
Comment (philippemnoel) [reply .pre-commit-config.yaml]: done

**Pull request merged**